### ISSUE TITLE
Lab Develop, Deploy, and debug a custom module on Azure IoT Edge with VS Code - Debugging is not working

### ISSUE THREAD

**Issue by ClarkDRC** [MicrosoftLearning/AZ-220-Microsoft-Azure-IoT-Developer#48]
# Module: 00
#### Step: 00

Description of issue

Repro steps:

1. Follow lab up until exercise 5 Task 2. Get the following error.
 
![Capture1](https://user-images.githubusercontent.com/49040027/88312546-41fc0f00-cce0-11ea-8abf-ecf1e46d3a7a.PNG)

**Comment by darenm** [comment]
@ClarkDRC I just ran through the lab and did not encounter the error, so I suspect you may have a local configuration issue. You can try uninstalling and reinstalling the Visual Studio IoTEdge extension and the other prerequisites and retry to see if this is a "one-off" issue. However, if the problem persists, I suggest you create an issue on the extension site: [microsoft/vscode-azure-iot-edge](https://github.com/microsoft/vscode-azure-iot-edge)

**Comment by MikeBazMSFT** [comment]
FYI @darenm I was able to reproduce this as well with exactly the same symptom, starting from zero in the LODS VM, following the Lab 3 install, then this lab (the Lab 3 install ends up taking quite some time in the LODS VM actually)... so I don't think it's completely one-off.

**Comment by ClarkDRC** [comment]
Did you use the LODs environment? I think it may be an issue with that. Michael B. verified that he got the error also.

**Comment by darenm** [comment]
Alas no - I don't presently have access to the LODS env - but it does tend to confirm my assumption that it is environmental. During the course development we ran into a number of issues with timeouts and overall performance with the IoTEdge simulator in the LODS VM. Although this doesn't specifically seem to be timeout related, it could be related to the installed software versions. Is the VM still running python 3.7 vs 3.8? Are you able to update the VS Azure IoT extensions to the latest version?

**Comment by MikeBazMSFT** [comment]
in my case it was from new so it was 3.8 (even though the lab instructions are a little contradictory on this in Lab 3) and the most recent extensions - looks like neither is actually in the LODS VM so you have to do the manual install steps.

**Comment by darenm** [comment]
OK - looks like something we will need to dig into further. As a key member of the team is out on vacation until next week, there will be a bit of a delay. If you have a chance, I would be curious to know if either of you are able to complete it using an "actual" PC as opposed to a VM.

**Comment by MikeBazMSFT** [comment]
ok, so I did this on my local machine, which was pretty clean at the start (new load less than a month ago), and it failed, but it failed in a different way.  Two observations:

1. On my machine, the setup step attempts to run in PowerShell; in the VM, it's attempting to run in WSL.  I have no idea how Code is making this decision one way or the other.
1. It still fails in PowerShell, but with a TLS error while running the Docker build.  I suspect this is the typical "set .NET TLS version" thing that comes up a lot - that would require a step in the setup or, honestly feels more like a bug in the tooling since _we_ shouldn't have to deal with it.

**Comment by ClarkDRC** [comment]
In the LODs environment I verified that iotedgehubdev.exe was installed in the correct folder. I also was able to start in using the cmd window. It seems to be an issue with VS Code recognizing it.
Unfortunately, I will have to skip this lab (running the class next week) until we can get it resolved.

**Comment by darenm** [comment]
So it looks like some issues that need to be raised on the vscode extension site. If we have some simple workarounds, we can include them in the lab for now... When we worked on this lab originally, it was 50/50 whether we kept it in due to performance on the VM as well as the time-out issue.

**Comment by ClarkDRC** [comment]
I am open to any work arounds we can come up with.

**Comment by ClarkDRC** [comment]
Looks like it will work with VS 2019.

**Comment by darenm** [comment]
@ClarkDRC Did you create an issue on the VS Ext site?

**Comment by ClarkDRC** [comment]
No.

**Comment by ClarkDRC** [comment]
What is the status on this?

**Comment by ChrisHowd** [comment]
@ClarkDRC Delay getting access, but will work to resolve issue next week.

**Comment by ClarkDRC** [comment]
Can we also get Docker pre-installed on the LODs vm for this lab?

**Comment by MikeBazMSFT** [comment]
That needs to be done through an LODS support ticket...each ALH has responsibility for their setup.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by darenm** [comment]
@ClarkDRC We have gained access to the LODS environment, replicated the issue and captured sufficient information to raise an [issue](https://github.com/microsoft/vscode-azure-iot-edge/issues/556) with the Visual Studio Code IoT Edge Extension team and are also escalating via the LODS team. We also tried the lab in the ExpressLabs VM environment and were able to complete the lab successfully. It should be noted that the ExpressLabs environment is using an earlier version of Python and Docker - we will investigate whether the ExpressLabs environment also fails using the latest versions.
